src/Slices/ServiceInventory/UI/Presenters/InstanceExpertActionPresenter.ts

```diff
@@ -0,0 +1,45 @@
+import React, { ReactElement } from "react";
+import { ServiceModel } from "@/Core";
+import { ActionPresenter, ServiceInstanceForAction } from "@/UI/Presenters";
+import { ExpertActions } from "../Components/ExpertActions";
+
+export class InstanceExpertActionPresenter implements ActionPresenter {
+  constructor(
+    private readonly instances: ServiceInstanceForAction[],
+    private readonly serviceEntity: ServiceModel
+  ) {}
+
+  private getInstanceForId(id: string): ServiceInstanceForAction | undefined {
+    return this.instances.find((instance) => instance.id === id);
+  }
+
+  getForId(id: string): ReactElement | null {
+    const instance = this.getInstanceForId(id);
+    if (typeof instance === "undefined") return null;
+    const states = this.serviceEntity.lifecycle.states.map(
+      (state) => state.name
+    );
+    return React.createElement(ExpertActions, {
+      instance,
+      targets: states,
+    });
+  }
+
+  isTransferDisabled(
+    id: string,
+    transferType: "on_update" | "on_delete"
+  ): boolean {
+    const instance = this.getInstanceForId(id);
+    if (typeof instance === "undefined") {
+      return false;
+    }
+    // If the action is allowed, there is a corresponding transfer in the lifecycle,
```

Comment:
Typically, this type of comment is best done as docstring. That way, the information is made available in the tooltip on hovering over the method, or with the intelisense.